components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/signup/UserSelfRegistrationManager.java

```diff
@@ -768,6 +768,35 @@ private UserRecoveryData validateSelfRegistrationCode(String code, String verifi
         return recoveryData;
     }
 
+    /**
+     * Introspects self registration confirmation code details without invalidating it.
+     * Does not triggering notification events or update user claims.
+     *
+     * @param skipExpiredCodeValidation   Skip confirmation code validation against expiration.
+     * @param code                      Confirmation code.
+     * @return UserRecoveryData           Data associated with the provided code, including related user and scenarios.
+     * @throws IdentityRecoveryException  Error validating the confirmation code
+     */
+    private UserRecoveryData introspectSelfRegistrationCode(String code, boolean skipExpiredCodeValidation)
+            throws IdentityRecoveryException {
+        Utils.unsetThreadLocalToSkipSendingEmailVerificationOnUpdate();
```

Comment:
is this required?

Reply:
where did we set this thread local?

Reply:
Yes we can remove this, this is set in https://github.com/wso2-extensions/identity-governance/blob/master/components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/signup/UserSelfRegistrationManager.java#L753 (In  VERIFY_EMAIL recovery step)